zhaquirks/tuya/ts1201.py

```diff
@@ -0,0 +1,524 @@
+                "seq": t.uint16_t,
+                "zero1": t.uint16_t,
+            },
+            direction=foundation.Direction.Server_to_Client,
+            is_manufacturer_specific=True,
+        )
+        receive_ir_frame_05: Final = foundation.ZCLCommandDef(
+            id=0x05,
+            schema={
+                "seq": t.uint16_t,
+                "zero": t.uint16_t,
+            },
+            direction=foundation.Direction.Server_to_Client,
+            is_manufacturer_specific=True,
+        )
+
+    class ClientCommandDefs(BaseCommandDefs):
+        """Client command definitions."""
+
+        resp_ir_frame_03: Final = foundation.ZCLCommandDef(
+            id=0x03,
+            schema={
+                "zero": t.uint8_t,
+                "seq": t.uint16_t,
+                "position": t.uint32_t,
+                "msgpart": t.LVBytes,
+                "msgpartcrc": t.uint8_t,
+            },
+            direction=foundation.Direction.Client_to_Server,
+            is_manufacturer_specific=False,
+        )
+        resp_ir_frame_05: Final = foundation.ZCLCommandDef(
+            id=0x05,
+            schema={
+                "seq": t.uint16_t,
+                "zero": t.uint16_t,
+            },
+            direction=foundation.Direction.Server_to_Client,
+            is_manufacturer_specific=True,
+        )
+
+    def handle_cluster_request(
+        self,
+        hdr: foundation.ZCLHeader,
+        args: list[Any],
+        *,
+        dst_addressing: Optional[
+            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
+        ] = None,
+    ):
+        """Handle a cluster request."""
+
+        # send default response, so avoid repeated zclframe from device
+        if not hdr.frame_control.disable_default_response:
+            self.debug("Send default response")
+            self.send_default_rsp(hdr, status=foundation.Status.SUCCESS)
+
+        if hdr.command_id == self.ServerCommandDefs.receive_ir_frame_00.id:
+            self.debug("hdr.command_id == 0x00")
+
+            self.current_position = 0
+            self.ir_msg.clear()
+            self.msg_length = args.length
+
+            cmd_01_args = {
+                "zero": 0,
+                "seq": args.seq,
+                "length": args.length,
+                "unk1": args.unk1,
+                "clusterid": args.clusterid,
+                "unk2": args.unk2,
+                "cmd": args.cmd,
+                "unk3": args.unk3,
+            }
+            self.create_catching_task(
+                super().command(0x01, **cmd_01_args, expect_reply=True)
+            )
+            cmd_02_args = {"seq": args.seq, "position": 0, "maxlen": 0x38}
+            self.create_catching_task(
+                super().command(0x02, **cmd_02_args, expect_reply=True)
+            )
+        elif hdr.command_id == self.ServerCommandDefs.receive_ir_frame_01.id:
+            self.debug("IR-Message-Code01 received, sequence: %s", args.seq)
```

Comment:
Ok, one last thing and I think we can try this afterwards.
I think `self.debug` uses the zigpy logger, so it's unclear where the message is coming from (a quirk).
Since we're logging quite a few messages, we should consider changing this.

We can either just include something like "Tuya IR: " as a prefix for the message every time, or create a custom logger `_LOGGER = logging.getLogger(__name__)` in this class and use that everywhere `_LOGGER.debug("...")`. That makes it clear it's `zhaquirks.tuya` that's logging these messages. We may also want to include the NwkId or IEEE (whatever you prefer), so it's clear which exact device is logging those messages.

Reply:
Should be addressed with https://github.com/zigpy/zha-device-handlers/pull/2336/commits/dd490a917439dd2516d3331bdab9595208e2b540